### PR TITLE
Add recipe for traffic library

### PR DESCRIPTION
**Pull request by xoolive**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xoolive** [comment]
@conda-forge/help-python I don't understand the dependency issue raised on MacOS and Windows (not Linux). I looked at the "conflicts" raised and they do not look conflicting to me. 

Any idea what could be going wrong?

**Comment by xoolive** [comment]
Hello @conda-forge/help-python

The win_64 and osx_64 recipes fail with dependency conflicts which do not look like conflicts to me. I think I need help/hints on this one.

**Comment by xoolive** [comment]
@conda-forge/help-python I am looking at other (merged) pull requests and it seems this error has been quite common with `win_64` target. Here I have `osx_64` failing as well with the same `DependencyError` but the common rule of thumb seems to be to accept `noarch` recipes as long as they are fine with `linux_64`.

I will be waiting for your green light (or advice) then...

File: recipes/traffic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - pip
+  run:
+    - python >=3.7
+    - numpy >=1.18
+    - scipy
+    - matplotlib-base
+    - pandas >=1.0.0
+    - pyproj >=2.0
+    - cartes >=0.2
+    - Shapely
```
Comment (chrisburr) [inline recipes/traffic/meta.yaml:28]: Conda package names are always lowercase
```suggestion
    - shapely
```

File: recipes/traffic/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - requests
+    - requests_pkcs12
+    - appdirs
+    - paramiko
+    - typing_extensions
+    - altair
+    - ipywidgets
+    - ipyleaflet
+    - tqdm >=4.28
+    - pyModeS >=2.9
```
Comment (chrisburr) [inline recipes/traffic/meta.yaml:38]:
```suggestion
    - pymodes >=2.9
```

**Comment by chrisburr** [comment]
The recipe for `requests_pkcs12` is buggy:
```
$ CONDA_SUBDIR=osx-64 mamba create --name test 'requests_pkcs12'
Encountered problems while solving:
  - nothing provides libgcc-ng >=4.9 needed by requests_pkcs12-1.0.4-pyh470a237_0
```
I'll fix it in https://github.com/conda-forge/requests_pkcs12-feedstock/pull/12

File: recipes/traffic/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - pyarrow
+    - beautifulsoup4
+    - lxml
+    - metar
+
+test:
+  source_files:
+    - traffic
+  requires:
+    - pytest
```
Comment (chrisburr) [inline recipes/traffic/meta.yaml:49]: It looks like you're not actually running the unit tests:
```suggestion
  requires:
```

Comment (xoolive) [reply recipes/traffic/meta.yaml:49]: Indeed the tests are running already many times on the github actions, so I'll leave it as you suggest.

**Comment by xoolive** [comment]
> I'll fix it in [conda-forge/requests_pkcs12-feedstock#12](https://github.com/conda-forge/requests_pkcs12-feedstock/pull/12)

Thank you @chrisburr so I should just wait for the fix and update the dependency number in the `meta.yaml` when ready?

**Comment by chrisburr** [comment]
> I should just wait for the fix and update the dependency number in the meta.yaml when ready?

No need to do anything, as this is noarch it'll just start working on macOS once the fixed `requests_pkcs12` build is available.

**Comment by xoolive** [comment]
@chrisburr it seems there is a dll import issue on pillow for friends trying to install on Windows. Taking pillow from regular conda seems to fix this issue (but then redistributable kit etc.)

Would it be a similar issue?

**Pull request merged**